controllers/sink.go

```diff
@@ -49,33 +45,33 @@ func (r *SinkReconciler) ObserveSinkStatefulSet(ctx context.Context, req ctrl.Re
 	}, statefulSet)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			r.Log.Info("sink is not ready yet...")
-			return nil
+			r.Log.Info("sink statefulset is not found...")
+		} else {
+			sink.Status.Conditions[v1alpha1.StatefulSet] = condition
+			return err
 		}
-		return err
+	} else {
+		// statefulset created, waiting it to be ready
+		condition.Action = v1alpha1.Wait
 	}
 
 	selector, err := metav1.LabelSelectorAsSelector(statefulSet.Spec.Selector)
 	if err != nil {
 		r.Log.Error(err, "error retrieving statefulSet selector")
-		return err
+	} else {
+		sink.Status.Selector = selector.String()
 	}
-	sink.Status.Selector = selector.String()
 
 	if *statefulSet.Spec.Replicas != *sink.Spec.Replicas {
-		condition.Status = metav1.ConditionFalse
 		condition.Action = v1alpha1.Update
-		sink.Status.Conditions[v1alpha1.StatefulSet] = condition
-		return nil
 	}
 
 	if statefulSet.Status.ReadyReplicas == *sink.Spec.Replicas {
 		condition.Action = v1alpha1.NoAction
 		condition.Status = metav1.ConditionTrue
-	} else {
-		condition.Action = v1alpha1.Wait
 	}
-	sink.Status.Replicas = *statefulSet.Spec.Replicas
+
+	sink.Status.Replicas = statefulSet.Status.ReadyReplicas
```

Comment:
statefulset's status has `replicas` and `readyReplices`, they are not equivalent.

we should not assign the `readyReplicas` to this replicas.

https://v1-18.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#statefulsetstatus-v1-apps

Reply:
I thought we want to now how many pod is ready instead of now many pod we want in status
or how about we change it to string of format `ReadyReplicas/Replicas`

Reply:
NVM, seems it's for HPA, not for user to observe. need to be statefulSet.Spec.Replicas